src/Http/Http.Abstractions/src/Routing/EndpointHttpContextExtensions.cs

```diff
@@ -34,29 +33,7 @@ public static class EndpointHttpContextExtensions
     public static void SetEndpoint(this HttpContext context, Endpoint? endpoint)
     {
         ArgumentNullException.ThrowIfNull(context);
-
-        var feature = context.Features.Get<IEndpointFeature>();
-
-        if (endpoint != null)
-        {
-            if (feature == null)
-            {
-                feature = new EndpointFeature();
-                context.Features.Set(feature);
-            }
-
-            feature.Endpoint = endpoint;
-        }
-        else
-        {
-            if (feature == null)
-            {
-                // No endpoint to set and no feature on context. Do nothing
-                return;
-            }
-
-            feature.Endpoint = null;
-        }
+        context.Endpoint = endpoint;
```

Comment:
Do I understand correctly that this is expected to update the `IEndpointFeature` appropriately?

Reply:
Yes, setting context.Endpoint = endpoint; will appropriately update the IEndpointFeature.Endpoint. The HttpContext.Endpoint property directly interacts with the IEndpointFeature, ensuring that any updates are synchronized between the two.